### PR TITLE
Implement new design for the Plugin Help Page.

### PR DESCRIPTION
**Pull request by qhuynh96**
* Implements new design for Plugin Help page. [Docs](https://docs.google.com/document/d/1oN2tqKzf_07MRkof1KTMAdmVk-BNB2NkvdiTyhmu47o/edit?usp=sharing)
* The page use [GoogleChrome/Polyfill](https://github.com/GoogleChrome/dialog-polyfill) libarary to enable dialog feature.
* All internal commands and external commands merged into a same table which flag by the [external icon](https://material.io/icons/#ic_link)

* Screenshots:

unset
unset 1

### CODE REVIEW

**Comment by qhuynh96** [comment]
/area prow

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by BenTheElder** [comment]
Maybe `s/example/examples/` on the header? this looks pretty snazzy!

**Comment by BenTheElder** [comment]
Discussed offline, we should fold external/normal plugins into one table and just use some annotation on the entry to note that it's external. Possibly https://material.io/icons/#ic_call_made (@qhuynh96 and I just looked through the icons).

**Comment by cjwagner** [comment]
Should we make the starred commands appear at the top of the list?

**Comment by BenTheElder** [comment]
> Should we make the starred commands appear at the top of the list?

I think that would help.

**Comment by BenTheElder** [comment]
/hold
Discussed offline, this also unfortunately depends on [`<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) under the hood which is not widely supported.

**Comment by qhuynh96** [comment]
/hold cancel
@BenTheElder  discussed offline about GoogleChrome/polyfill alternate

**Comment by qhuynh96** [comment]
@stevekuznetsov  if you have time, can you have a look on this?

**Comment by stevekuznetsov** [comment]
Looking now -- is there a deployment somewhere I can poke around in?

**Comment by stevekuznetsov** [comment]
Yeah when the diff is so large GitHub won't render it by default... would be nice to have a PoC to interact with.

File: prow/cmd/deck/static/plugin-help-script.js [hunk 1]
```diff
@@ -4,215 +4,366 @@ const closedArrow = "\u25b6 ";
 const openedArrow = "\u25bc ";
```
Comment (stevekuznetsov) [inline prow/cmd/deck/static/plugin-help-script.js]: Might be nice to get your formatter to follow current style or do formatting pass as a separate commit/diff

Comment (qhuynh96) [reply prow/cmd/deck/static/plugin-help-script.js]: What is our current style? I will config my formatter to follow the style. Currently, I am using 2 spaces and wrap at 80.

Comment (stevekuznetsov) [reply prow/cmd/deck/static/plugin-help-script.js]: I would just compare what the commit before had ... looks like four spaces for indentation and no line width max that I can tell at all.

**Comment by qhuynh96** [comment]
@stevekuznetsov  The back-end changes checked in so you can fetch my branch and ./runlocal

File: prow/cmd/deck/static/plugin-help-script.js (hunk 1 above)
Comment (qhuynh96) [reply prow/cmd/deck/static/plugin-help-script.js]: Fixed indent and wrap

**Comment by qhuynh96** [comment]
Cherry pick snowflake Xref #6398 
/retest

File: prow/cmd/deck/static/dialog-polyfill.css [hunk 2]
```diff
@@ -0,0 +1,37 @@
+dialog {
```
Comment (BenTheElder) [inline prow/cmd/deck/static/dialog-polyfill.css]: does this need to be a separate stylesheet?

File: prow/cmd/deck/static/dialog-polyfill.js [hunk 3]
```diff
@@ -0,0 +1,738 @@
+(function() {
```
Comment (BenTheElder) [inline prow/cmd/deck/static/dialog-polyfill.js]: license? I suspect this is https://github.com/GoogleChrome/dialog-polyfill which has [a BSD license](https://github.com/GoogleChrome/dialog-polyfill/blob/master/LICENSE) which requires redistributing the license.

File: prow/cmd/deck/static/dialog-polyfill.css (hunk 2 above)
Comment (qhuynh96) [reply prow/cmd/deck/static/dialog-polyfill.css]: it's from GoogleChrome/dialog-polify. I will add the license to it

Comment (BenTheElder) [reply prow/cmd/deck/static/dialog-polyfill.css]: We should still probably inline this into the prow stylesheet with a comment, this doesn't seem worth another request to load?

Comment (qhuynh96) [reply prow/cmd/deck/static/dialog-polyfill.css]: sure, I will do it tmr

**Comment by stevekuznetsov** [comment]
Looks really good. Only comment is that the first two columns are not super useful right now as they don't present different information. I like the "examples" column a lot more than the "command one" as well.

screenshot from 2018-01-26 14-15-50

**Comment by stevekuznetsov** [comment]
/lgtm
/hold

Un-hold if you want to tackle the columns in a follow-up.

**Pull request merged**